### ISSUE TITLE
Users with space in their name may not be reachable for mentions

### ISSUE THREAD

**Issue by pgiraud** [hotosm/osm-tasking-manager2#369]
In http://tasks.hotosm.org/project/666#task/46, @harry-wood could not mention user "sea hik" in the comments.

**Comment by pyrog** [comment]
This issue seem to be solved :smile: 

But there are a similar issues with:
 1. names with accentuated chars or unicode:
* [Léo_M](http://tasks.dev.hotosm.org/user/L%C3%A9o_M)
* [陳柏瑋](http://tasks.dev.hotosm.org/user/%E9%99%B3%E6%9F%8F%E7%91%8B)

 They raise an error when you click on the user link:
_"**An error occurred**. You can try to resolve the error by manually logging out"_

 2. Some user name are not recognize in the contribute tab of a task.<br/>
regular expression in [osmtm/views/task.py#L228](https://github.com/hotosm/osm-tasking-manager2/blob/676e2530ad97218f0061baebf5b80f2f24f399f2/osmtm/views/task.py#L228) don't work
```python
p = re.compile(ur'((?<=@)\w+|\[.+?\])')
```

This one seem to work fine (see http://regexr.com/3a5sc)
```python
ur'(?:@\[(.+)\])|(?:@(.\w+))'
```

* **test @!i!** should be **test [@!li!](http://tasks.dev.hotosm.org/user/!i!)**<br/>[![test i](https://cloud.githubusercontent.com/assets/2158081/5628614/1ca69162-95aa-11e4-9078-fa9285d7eb99.png)](http://tasks.dev.hotosm.org/project/611#task/1027)
* @ちゃい
* Léo_M
* <0174

**Comment by harry-wood** [comment]
You said it was solved, but didn't say what the solution was!

For people looking to do this, you can mention a user if they have a space in their name, by putting square brackets. For example  @[Lisa Marie Owen]    ...see this working on the site here: http://tasks.hotosm.org/project/827#task/15

**Comment by pyrog** [comment]
> You said it was solved, but didn't say what the solution was!

Yes, because I'am not the author of either the issue and either of the fix. And I didn't say that it's solved, just that it *"seem"* to be…:smile:
